core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -596,7 +597,11 @@ sealed abstract class P2WPKHSigner
                              unsignedTx,
                              isDummySignature,
                              spendingInfoToSatisfy)
-        sig = sigComponent.scriptSignature.signatures.head
+        sig = sigComponent
+          .asInstanceOf[WitnessTxSigComponent]
+          .witness
+          .asInstanceOf[script.P2WPKHWitnessV0]
+          .signature
```

Comment:
It'd be safer to use a `match` here and throw errors when you don't get the expected types